src/texts/sdf/sdf.js

```diff
@@ -117,36 +117,177 @@ export default class {
     // TODO: this code is not intuitive, we can write better
     return this.atlas.texture;
   }
+  
+  // function to align text left, right or center
+  alignText(alignment, text, x, y, width, height, markDirty, widthArray, wordWidth) {
 
+    // x and y are the clipspace co-ordinates between 0 and 1
+   // dx and dy shifts the position of label w.r.t possibly node
+   // (TODO: dx and dy are calculated w.r.t what is not clear , please clear it if you find out)
+    
+   if(!( alignment === 'right' || alignment === 'left' || alignment === 'middle')) {
+    alignment = 'left';
+    }
+    const textArray = text.split(" ");
+    let dx=0;
+
+    // dy positioned so that if y< 0.5 i.e. for lower half of canvas it's length should increase dynamically 
+    // so that it characters don't go outside of canvas, also for y>0.5 , it's constant at a particular height
+    // so that it for y===1 it doesn't go beyond the canvas
 
+    let dy = y <= 0.5 ? height/3*(textArray.length-1) : -height/3;
+    // "ret" must be the return object. "ret" is always the return object
+    let ret = [];
+
+    switch (alignment) {
+      case 'left' :
+        dx = x <= 0.5 ? 0: -width;
+        
+        for (var i=0;i<text.length;i++) {
+
+          // changing line here when it encounters any whitespace character
+
+          if ((text[i] === ' ') && (i != 0 || i != text.length-1) ) {
+            dx = x <= 0.5 ? 0 : -width ;
+            dy = dy-Math.floor(height/3);
+          } else {
+            const char = this._getChar(text[i], markDirty);
+            const rect = char.rect || {};
+            let horiBearingX = 3;
+            dx += horiBearingX;
+     
+            // rect.x rect.w rect.h rect.y are all atlas widths heigths x y positions etc 
+            ret.push({
```

Comment:
this part looks like a lot of the duplication of code. Is it possible wrap this part of the code as a function?

Reply:
Sure, did it!